common/src/subgraph_client/client.rs

```diff
@@ -93,14 +94,19 @@ impl DeploymentDetails {
             status_url: Some(Url::parse(graph_node_status_url)?),
             query_url: Url::parse(graph_node_base_url)?
                 .join(&format!("subgraphs/id/{deployment}"))?,
+            query_auth_token: None,
         })
     }
 
-    pub fn for_query_url(query_url: &str) -> Result<Self, anyhow::Error> {
+    pub fn for_query_url(
+        query_url: &str,
+        query_auth_token: Option<String>,
+    ) -> Result<Self, anyhow::Error> {
         Ok(Self {
             deployment: None,
             status_url: None,
             query_url: Url::parse(query_url)?,
+            query_auth_token,
         })
     }
```

Comment:
I think it would be better if we have a new function `fn for_query_url_with_token(query_url: &str, query_auth_token: Option<String>)` instead of overwriting the current `for_query_url` function.